core/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -197,14 +206,97 @@ public XContentBuilder innerToXContent(XContentBuilder builder, Params params) t
             builder.field(Fields.TERMINATED_EARLY, isTerminatedEarly());
         }
         RestActions.buildBroadcastShardsHeader(builder, params, getTotalShards(), getSuccessfulShards(), getFailedShards(), getShardFailures());
-        internalResponse.toXContent(builder, params);
+        internalResponse.hits.toXContent(builder, params);
+        if (internalResponse.aggregations != null) {
+            internalResponse.aggregations.toXContent(builder, params);
+        }
+        if (internalResponse.suggest != null) {
+            internalResponse.suggest.toXContent(builder, params);
+        }
+        if (internalResponse.profileResults != null) {
+            internalResponse.profileResults.toXContent(builder, params);
+        }
         return builder;
     }
 
+    public static SearchResponse fromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token;
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+        String currentFieldName = null;
+        InternalSearchHits hits = null;
+        boolean timedOut = false;
+        Boolean terminatedEarly = null;
+        long tookInMillis = 0;
+        int successfulShards = 0;
+        int totalShards = 0;
```

Comment:
shall we use -1 as default values, just to make sure not to confuse them with actual values that get returned?